src/Refactoring-Core/RBPullUpMethodRefactoring.class.st

```diff
@@ -50,25 +50,30 @@ RBPullUpMethodRefactoring class >> pullUp: selectorCollection from: aClass to: a
 
 { #category : 'preconditions' }
 RBPullUpMethodRefactoring >> applicabilityPreconditions [
+	" Check that all selectors are defined in `class`, 
+	and that `class` is in the hierarchy of subclasses of `targetSuperclass` "
 
 	^ {
-		  (RBCondition hasSuperclass: class).
 		  (ReDefinesSelectorsCondition new
 			   definesSelectors: selectors
 			   in: class).
-		  (RBCondition withBlock: [
-			   self checkClassVars.
-			   self checkSuperclass.
-			   self checkSuperMessages.
-			   true ]) }
+		  (ReClassHasSubclassesCondition new
+			   class: targetSuperclass;
+			   subclassesList: { class name }).
+		self preconditionNoOverrides.
+		  self preconditionNoSupersendsSent.
+		  self preconditionNoSupersendsReceived }
 ]
```

Comment:
@balsa-sarenac I understand your suggestion, however I deliberately removed both preconditions from `applicabilityPreconditions` and put them in `breakingChangePreconditions`. I did this because failedApplicabilityPreconditions interrupt the refactoring execution, while both `preconditionNoReferencesToInstVars` and `preconditionNoReferencesToSharedVars` should not prevent the execution of the refactoring because they can be automatically fixed. 

The modification I propose makes the following two tests fail: 
`RBPullUpMethodParametrizedTest>>testPullUpReferencesInstVar` 
`RBPullUpMethodParametrizedTest>>testWarnDoesntPullUpReferencesInstVar` 
Therefore if we decide to go the way I propose, these tests should be modified. 

What do you think?

Reply:
Feel free to change them. I guess they should raise `RBRefactoringWarning` now instead of `RBRefactoringError`?